File: leetcode/easy/count-elements-with-strictly-smaller-and-greater-elements.py
# https://leetcode.com/problems/count-elements-with-strictly-smaller-and-greater-elements/

class Solution:
    def countElements(self, nums: List[int]) -> int:
        ''''
        if len(nums) < 2: return 0
        mn, mx, num_min, num_max = nums[0], nums[0], 1, 1
        for i in range(1, len(nums), 1):
            if nums[i] < mn:
                mn = nums[i]
                num_min = 1
            elif nums[i] == mn: num_min += 1
            if nums[i] > mx:
                mx = nums[i]
                num_max = 1
            elif nums[i] == mx: num_max += 1
        return 0 if mn == mx else len(nums) - num_min - num_max
        '''

        mn, mx = min(nums), max(nums)
        return sum(1 for x in nums if mn < x < mx)

File: leetcode/easy/count-elements-with-strictly-smaller-and-greater-elements.ts
// https://leetcode.com/problems/count-elements-with-strictly-smaller-and-greater-elements/

// MINE
/*
function countElements(nums: number[]): number {
    const max = Math.max(...nums), min = Math.min(...nums)
    let ans = 0
    for(let n of nums) if(n < max && n > min) ++ans
    return ans
};
*/

// ANSWER
function countElements(nums: number[]): number {
  if (nums.length < 2) return 0
  let min = nums[0],
    max = nums[0],
    numMin = 1,
    numMax = 1
  for (let i = 1; i < nums.length; ++i) {
    if (nums[i] < min) {
      min = nums[i]
      numMin = 1
    } else if (nums[i] === min) ++numMin
    if (nums[i] > max) {
      max = nums[i]
      numMax = 1
    } else if (nums[i] === max) ++numMax
  }
  return min === max ? 0 : nums.length - numMin - numMax
}
